packages/react-dialog/Spec.md

```diff
@@ -2,62 +2,303 @@
 
 ## Background
 
-_Description and use cases of this component_
+A dialog is a window overlaid on top of the page, and it is used to inform users of critical information, require a decision or to complete a task.
 
 ## Prior Art
 
-_Include background research done for this component_
+- All mentions of v7 or v8 refer to Fabric - `@fluentui/react` ([docsite](https://developer.microsoft.com/en-us/fluentui#/))
+- All mentions of v0 refer to Northstar - `@fluentui/react-northstar` ([docsite](https://fluentsite.z22.web.core.windows.net/))
 
-- _Link to Open UI research_
-- _Link to comparison of v7 and v0_
-- _Link to GitHub epic issue for the converged component_
+- [Github epic](https://github.com/microsoft/fluentui/issues/20953)
+- [Open UI Research](https://open-ui.org/components/dialog.research)
+- Dialogs in 3rd party UI systems:
+  - [Carbon](https://react.carbondesignsystem.com/?path=/docs/components-modal--default)
+  - [Chakra UI](https://chakra-ui.com/docs/overlay/modal)
+  - [FAST](https://explore.fast.design/components/fast-dialog)
+  - [Material UI](https://mui.com/components/dialogs/)
+  - [Radix](https://www.radix-ui.com/docs/primitives/components/dialog)
+  - [Reach UI](https://reach.tech/dialog/)
+  - [Reakit](https://reakit.io/docs/dialog/)
+    - [Ariakit](https://github.com/reakit/reakit/tree/v2)
+  - [Spectrum](https://react-spectrum.adobe.com/react-spectrum/Dialog.html)
 
-## Sample Code
+### Comparison between v0 and v8
+
+Note that the below code samples are not meant to be complete, but to highlight differences between the two libraries. Please refer to official docsites for actual API references.
+
+#### v8
+
+In v8 there are Dialog and Modal components which are relevant to the Dialog component for v9. The Dialog component was intended to be used primarily for confirming actions, whereas Modal component was intended to be used for lengthy content that may contain forms and other controls. This spec will only cover the comparison to the Dialog component.
+
+The visibility of the dialog is controlled through the `hidden` prop whose its value should be a react state boolean provided from the consumer.
+
+[Documentation for v8 Dialog](https://developer.microsoft.com/en-us/fluentui#/controls/web/dialog)
+
+Sample code:
+
+```jsx
+  <DefaultButton secondaryText="Opens the Sample Dialog" onClick={toggleHideDialog} text="Open Dialog" />
+
+  <Dialog
```

Comment:
I would advise you look into how to revert focus to the last focused element. `Popover` which has similar focus behaviour to `Dialog` follows the trigger pattern which makes this slightly easier.

I don't have a huge preference for `Dialog` to follow the trigger pattern. Honestly the way you have it as a single component is quite simple (which I like). However reverting focus is a really important thing and unless we think about how to do that with the current proposed API, it might be complicated later on

Reply:
feel free to reach out if you want to talk more about this